x/ibc-rate-limit/contracts/rate-limiter/src/state.rs

```diff
@@ -17,16 +17,19 @@ pub enum FlowType {
     Out,
 }
 
-/// A Flow represents the transfer of value through an IBC channel duriong a
+/// A Flow represents the transfer of value through an IBC channel during a
 /// time window.
 ///
 /// It tracks inflows (transfers into osmosis) and outflows (transfers out of
 /// osmosis).
 ///
 /// The period_end represents the last point in time that for which this Flow is
 /// tracking the value transfer.
+/// TODO: Document that time windows are not rolling windows, but instead discrete repeating windows.
+/// This is a design decision chosen for gas reasons.
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema, Copy)]
 pub struct Flow {
+    // Q: Do we have edge case issues with inflow/outflow being u128, e.g. what if a token has super high precision.
```

Comment:
Highlighting question

Reply:
That's a really good question. Internally, cosmwasm uses `Uint128` ([bank queries](https://github.com/CosmWasm/cosmwasm/blob/9bf920b00b76f31f3c73ed3632cd07cbd6a94b7b/packages/std/src/results/cosmos_msg.rs#L61), for example): https://github.com/CosmWasm/cosmwasm/blob/main/packages/std/src/coins.rs#L10. So that's why I went with 128. 

Their implementation is basically `u128` with some [helpers](https://github.com/CosmWasm/cosmwasm/blob/main/packages/std/src/math/uint128.rs) attached.

It should be easy to change to larger representations though. Cosmwasm has `Uint256` and `Uint512`.